copy_user_fields.py

```diff
@@ -11,60 +11,78 @@
 
 
 def v1_get_unique_fields(fields: list) -> list:
+    """
+    # >>> fields = [{'name': 'f0', 'fieldType': 'STRING', 'contentType': 'METADATA', 'sort': True, 'facet': False}, \
+    # {'name': 'f0', 'fieldType': 'STRING', 'contentType': 'METADATA', 'sort': False, 'facet': True}, \
+    # {'name': 'f1', 'fieldType': 'STRING', 'contentType': 'METADATA'}]
+    # >>> v1_get_unique_fields(fields)
+    # [('f0', [{'name': 'f0', 'fieldType': 'STRING', 'contentType': 'METADATA', 'sort': True, 'facet': False}, {'name': 'f0', 'fieldType': 'STRING', 'contentType': 'METADATA', 'sort': False, 'facet': True}]), ('f1', [{'name': 'f1', 'fieldType': 'STRING', 'contentType': 'METADATA'}])]
+    """
     def get_fields_by_name() -> dict:
-        v1_fields_by_name = [(field['name'], field) for field in fields]
-        # deduplicate
-        return dict([(field[0],
-                      [f[1] for f in v1_fields_by_name if f[0] == field[0]])
-                     for field in v1_fields_by_name])
-
-    def validate_same_config(fields: list) -> bool:
-        previous = fields[0]
-        for i in range(1, len(fields)):
-            current = fields[i]
-            if not (previous['fieldType'] == fields[i]['fieldType'] and
-                    previous['contentType'] == fields[i]['contentType'] and
-                    previous['fieldQueries'] == fields[i]['fieldQueries'] and
-                    previous['freeTextQueries'] == fields[i]['freeTextQueries'] and
-                    previous['facet'] == fields[i]['facet'] and
-                    previous['multivalueFacet'] == fields[i]['multivalueFacet'] and
-                    previous['sort'] == fields[i]['sort'] and
-                    previous['displayField'] == fields[i]['displayField']):
-                print(
-                    f'SKIPPING FIELD {current["name"]}. Found fields in CloudV1 with the same name but different configurations: {fields}')
-                return False
-            previous = fields[i]
+        fields_by_name = dict()
+        for field in fields:
+            name = field['name']
+            if name in fields_by_name:
+                fields_by_name[name].append(field)
+            else:
+                fields_by_name[name] = [field]
+        return fields_by_name
+
+    def merge_fields_config(fields: list) -> dict:
+        new_field = {'fieldQueries': False, 'freeTextQueries': False, 'facet': False, 'multivalueFacet': False, 'sort': False, 'displayField': False}
+        mappings = list()
+        for field in fields:
+            mapping = {'name': field['name'], 'metadataName': field['metadataName'], 'sourceId': field['sourceId']}
+            for flag in new_field:
+                if type(new_field[flag]) == bool:
```

Comment:
No need for that check, but sure why not

Reply:
You are right, that's an artifact from a previous iteration